test/test_edit.py

```diff
@@ -0,0 +1,172 @@
+# This file is part of beets.
+# Copyright 2015, Adrian Sampson and Diego Moreda.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+from __future__ import (division, absolute_import, print_function,
+                        unicode_literals)
+import codecs
+
+from mock import Mock, patch
+from test._common import unittest
+from test.helper import TestHelper, control_stdin
+
+from beets import library
+from beetsplug.edit import EditPlugin
+
+
+class ModifyFileMocker(object):
```

Comment:
I have used this helper for bypassing the calls to the external editor (`EditPlugin.get_editor()`), in order to hopefully simplify the actual `test_xxx` methods. The reason for making it a class is to be able to pass some parameters (in short, the changes to be mimicked on the yaml files) to `overwrite_contents()` and `replace_contents()`, while still being able to use those functions as a side effect of `patch.object()`. I'm not that familiar with mock - maybe this can be achieved in a less verbose way?